src/platform_impl/linux/x11/util/cursor.rs

```diff
@@ -45,78 +47,15 @@ impl XConnection {
-        }
-    }
-
-    fn load_first_existing_cursor(&self, names: &[&[u8]]) -> ffi::Cursor {
-        for name in names.iter() {
-            let xcursor = self.load_cursor(name);
-            if xcursor != 0 {
-                return xcursor;
-            }
-        }
-        0
-    }
-
     fn get_cursor(&self, cursor: Option<CursorIcon>) -> ffi::Cursor {
         let cursor = match cursor {
             Some(cursor) => cursor,
             None => return self.create_empty_cursor(),
         };
 
-        let load = |name: &[u8]| self.load_cursor(name);
-
-        let loadn = |names: &[&[u8]]| self.load_first_existing_cursor(names);
-
-        // Try multiple names in some cases where the name
-        // differs on the desktop environments or themes.
-        //
-        // Try the better looking (or more suiting) names first.
-        match cursor {
-            CursorIcon::Alias => load(b"link\0"),
-            CursorIcon::Arrow => load(b"arrow\0"),
-            CursorIcon::Cell => load(b"plus\0"),
-            CursorIcon::Copy => load(b"copy\0"),
-            CursorIcon::Crosshair => load(b"crosshair\0"),
-            CursorIcon::Default => load(b"left_ptr\0"),
-            CursorIcon::Hand => loadn(&[b"hand2\0", b"hand1\0"]),
-            CursorIcon::Help => load(b"question_arrow\0"),
-            CursorIcon::Move => load(b"move\0"),
-            CursorIcon::Grab => loadn(&[b"openhand\0", b"grab\0"]),
-            CursorIcon::Grabbing => loadn(&[b"closedhand\0", b"grabbing\0"]),
-            CursorIcon::Progress => load(b"left_ptr_watch\0"),
-            CursorIcon::AllScroll => load(b"all-scroll\0"),
-            CursorIcon::ContextMenu => load(b"context-menu\0"),
-
-            CursorIcon::NoDrop => loadn(&[b"no-drop\0", b"circle\0"]),
-            CursorIcon::NotAllowed => load(b"crossed_circle\0"),
-
-            // Resize cursors
-            CursorIcon::EResize => load(b"right_side\0"),
-            CursorIcon::NResize => load(b"top_side\0"),
-            CursorIcon::NeResize => load(b"top_right_corner\0"),
-            CursorIcon::NwResize => load(b"top_left_corner\0"),
-            CursorIcon::SResize => load(b"bottom_side\0"),
-            CursorIcon::SeResize => load(b"bottom_right_corner\0"),
-            CursorIcon::SwResize => load(b"bottom_left_corner\0"),
-            CursorIcon::WResize => load(b"left_side\0"),
-            CursorIcon::EwResize => load(b"h_double_arrow\0"),
-            CursorIcon::NsResize => load(b"v_double_arrow\0"),
-            CursorIcon::NwseResize => loadn(&[b"bd_double_arrow\0", b"size_fdiag\0"]),
-            CursorIcon::NeswResize => loadn(&[b"fd_double_arrow\0", b"size_bdiag\0"]),
-            CursorIcon::ColResize => loadn(&[b"split_h\0", b"h_double_arrow\0"]),
-            CursorIcon::RowResize => loadn(&[b"split_v\0", b"v_double_arrow\0"]),
-
-            CursorIcon::Text => loadn(&[b"text\0", b"xterm\0"]),
```

Comment:
Was this "fallback" mechanism here important? It might be good to have a way to load the backup "xterm" cursor in the event that we fail to load the "text" cursor.

Reply:
I don't really care https://www.freedesktop.org/wiki/Specifications/cursor-spec/ . All modern themes have everything correct. The patch is also trivial and could be solved by renaming icon.